fhir/client.go

```diff
@@ -85,22 +101,48 @@ func (c *Client) NewSearchTypeRequest(resourceType string) (*http.Request, error
 // SearchTypeURL generates the URL for a search-type interaction of the given
 // resource type
 func (c *Client) SearchTypeURL(resourceType string) string {
-	return c.Base + "/" + resourceType
+	rel := &url.URL{Path: path.Join(c.baseURL.Path, resourceType)}
```

Comment:
Join isn't needed if the baseURl ends with a slash.

Reply:
Adjustment would be justified. However, I will remove this function entirely alongside the `NewSearchTypeRequest` function that uses it. This is because the latter is not used anywhere in the code. It's just not been marked by the IDE since it is a public function.

Reply:
ok